mzLib/FlashLFQ/FlashLfqEngine.cs

```diff
@@ -1967,7 +1973,11 @@ private void QuantifyIsobaricPeaks()
             double lastReportedProgress = 0;
             double currentProgress = 0;
 
-            var idGroupedBySeq = _allIdentifications
+            // Filter out the id with motif checking from the motif list we uploaded
+            var ids = SearchingTarget.TargetMotifs!= null?
+                _allIdentifications.Where(p => SearchingTarget.MotifFilter(p.ModifiedSequence)).ToList() : _allIdentifications;
+            // Group the IDs by their base sequence and monoisotopic mass -> isobaric peptide
+            var idGroupedBySeq = ids
```

Comment:
Please write the assumptions down. Isotracker only runs IF modified AND modification contains residue.